applications/FluidDynamicsApplication/custom_conditions/adjoint_monolithic_wall_condition.h

```diff
@@ -0,0 +1,418 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ \.
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+#if !defined(KRATOS_ADJOINT_MONOLITHIC_WALL_CONDITION_H_INCLUDED)
+#define KRATOS_ADJOINT_MONOLITHIC_WALL_CONDITION_H_INCLUDED
+
+// System includes
+#include <string>
+
+// External includes
+
+// Project includes
+#include "includes/condition.h"
+#include "includes/define.h"
+#include "includes/process_info.h"
+#include "includes/serializer.h"
+#include "utilities/adjoint_extensions.h"
+#include "utilities/indirect_scalar_fwd.h"
+
+// Application includes
+
+namespace Kratos
+{
+///@addtogroup FluidDynamicsApplication
+///@{
+
+///@name Kratos Classes
+///@{
+
+template <unsigned int TDim, unsigned int TNumNodes = TDim>
+class KRATOS_API(FLUID_DYNAMICS_APPLICATION) AdjointMonolithicWallCondition : public Condition
+{
+    class ThisExtensions : public AdjointExtensions
```

Comment:
What is the purpose of this container? As far as I see, it only owns a pointer to the father condition so couldn't the same behavior be achieved by passing the reference to the condition of interest (as well as the hardcoded variables in the source file) to an external utility with all the methods below? I think this would be easier and reusable for other adjoints (you could even make all these `static`)

Alternatively, it could make sense to have an intermediate layer between `Condition` and this called `AdjointCondition` to enhance the API with these.

(dismiss all these observations if these extensions are only to be used in here).

Reply:
The use of this container is to create `IndirectScalar` for values returned from `GetValuesVector`, `GetSecondDerivativesVector`, etc... The `GetValuesVector` gives a copy of the actual nodal values. But `AdjointSchemes` need to change them for some computations, and it does not know about the variables used in the element. So we use this intermediate container to provide these `IndectScalars` which give the reference to nodal values.

We are implementing all of these interfaces in every adjoint element and condition. But I also would like to have this more generalized. I'll keep that in mind and try to think of something in a separate PR.